unified-codes/apps/web/src/components/templates/Explorer.tsx

```diff
@@ -23,9 +29,31 @@ export interface ExplorerProps {
   onUpdateVariables: (variables: IExplorerVariables) => void;
 }
 
+const getStyles = (theme: ITheme) => {
+  const borderStyle = `1px solid ${theme.palette.divider}`;
+  return {
+    pagination: { backgroundColor: theme.palette.background.toolbar },
+    root: { backgroundColor: theme.palette.background.default, maxHeight: '100%', maxWidth: 900 },
+    searchBar: { paddingLeft: 15 },
+    table: {
+      marginTop: 5,
+      maxHeight: `calc(100vh - ${FOOTER_HEADER_HEIGHT}px)`,
+      overflowY: 'scroll' as overflow,
+      '& th': { backgroundColor: theme.palette.background.toolbar, fontWeight: 700 },
+      '& thead > tr': { borderBottom: borderStyle },
+      '& tr > th': { borderRight: borderStyle },
+      '& tr > td': { borderRight: borderStyle, borderBottom: 0 },
+      '& tr > td:last-child': { borderRight: 0 },
+      '& tr > th:last-child': { borderRight: 0 },
+      '& tr > td:first-child': { fontWeight: 700 },
+    },
+  };
+};
+
 export type Explorer = React.FunctionComponent<ExplorerProps>;
 
 export const ExplorerComponent: Explorer = ({
+  classes,
```

Comment:
As above, not too sure about prop drilling `classes` via the parent to the child when the styles can be imported directly (would this not end up with parents having to manage a really large number of `classes` as the tree got deeper?)

Reply:
oh, ok. I see where you're heading. I wouldn't want library components to be importing styles though. could I import at the web level and pass in to the lower components?

Reply:
I agree with not importing into library components (as that would just cause coupling as you imply), but since `Explorer` is a locally defined component, passing in `classes` from another locally defined component seems like potentially redundant prop drilling (but I'm happy to be wrong if this is an accepted pattern). 

I think its fair enough to have a generic `classes` and `style` prop for all components, and for parent classes to drill classes down the tree. But I think if we go this route we should include it in all our library components, so it's a consistent API.

Having said that, I think it's best to avoid relying too much on drilling generic props, mainly because I think it encourages coupling between components and could get messy quite quickly.

I think a good option is to have `classes` and `styles` for all components, but only for presentational stuff, and as much as possible only for the component itself, rather than specifying all the child styles in the parent. I think a better way to handle the issue of styling the subcomponents is to expose them as props directly, i.e. library components containing other library components expose their children as optional props. Since the children are also exported by the library, you can then wrap them in a closure with all the props you want them to get (again, though, ideally only presentational stuff, since their designed to be "dumb"). I've seen this pattern in a few places, and seems to have similar benefits as OOP dependency injection.

Here's an example with styling the header for an `EntityTable`:

```
const HeaderComponent = React.useCallback((props) => <EntityTableHeader {...props} style={YOUR_CODE_HERE}/>
...

<EntityTable
  headerComponent={HeaderComponent}
  rowComponent={RowComponent}
  ...
/>
```

I think this would also tidy up the library components a bit, which I think have grown to include conditional logic which I think we generally on the same page with wanting to avoid as much as possible (i.e. make the component as "dumb" as possible, push conditions up the tree as much as possible)?